addons/google_calendar/models/res_users.py

```diff
@@ -158,3 +165,38 @@ def check_calendar_credentials(self):
         client_secret = get_param('google_calendar_client_secret')
         res['google_calendar'] = bool(client_id and client_secret)
         return res
+
+    def check_synchronization_status(self):
+        res = super().check_synchronization_status()
+        credentials_status = self.check_calendar_credentials()
```

Comment:
Could you update this method to use the enhancements introduced by PR #95744? This way, we could more easily override the API key. Thank you!

```diff
diff --git addons/google_calendar/controllers/main.py addons/google_calendar/controllers/main.py
index 8c4fd5bec7d0..17ba1594e44a 100644
--- addons/google_calendar/controllers/main.py
+++ addons/google_calendar/controllers/main.py
@@ -5,7 +5,7 @@ from odoo import http
 from odoo.http import request
 from odoo.addons.google_calendar.utils.google_calendar import GoogleCalendarService
 from odoo.addons.calendar.controllers.main import CalendarController
-
+from odoo.addons.google_account.models.google_service import _get_client_secret
 
 class GoogleCalendarController(CalendarController):
 
@@ -62,8 +62,8 @@ class GoogleCalendarController(CalendarController):
     @http.route()
     def check_calendar_credentials(self):
         res = super().check_calendar_credentials()
-        get_param = request.env['ir.config_parameter'].sudo().get_param
-        client_id = get_param('google_calendar_client_id')
-        client_secret = get_param('google_calendar_client_secret')
+        client_id = request.env['google.service']._get_client_id('calendar')
+        ICP_sudo = request.env['ir.config_parameter'].sudo()
+        client_secret = _get_client_secret(ICP_sudo, 'calendar')
         res['google_calendar'] = bool(client_id and client_secret)
         return res
```